cpmpy/solvers/cpo.py

```diff
@@ -152,11 +152,16 @@ def solve(self, time_limit=None, **kwargs):
 
             All solver parameters are documented here: https://ibmdecisionoptimization.github.io/docplex-doc/cp/docplex.cp.parameters.py.html#docplex.cp.parameters.CpoParameters
 
-        """
+        """ 
 
-        # ensure all vars are known to solver
+        # ensure all vars are known to solver        
         self.solver_vars(list(self.user_vars))
 
+        # edge case, empty model, ensure the solver has something to solve
+        if not len(self.user_vars):
+            iv = intvar(1,1)
+            self += iv == 1
```

Comment:
We now have `self.add()` which looks a little more natural in these situations.
Also, no real need to make the iv variable itself. It can just be `self.add(intvar(1,1) == 1)`